mpi-proxy-split/test/run_all.py

```diff
@@ -0,0 +1,117 @@
+    'Ibcast_test1': {'itr': 100000000, 'ranks': 4, 'args': -1},
+    'Ibcast_test2': {'itr': 100000000, 'ranks': 4, 'args': -2},
+    'Initialized_test': {'itr': 100000000, 'ranks': 4},
+    'Irecv_test': {'itr': 12, 'ranks': 2},
+    'Isend_test': {'itr': 12, 'ranks': 2},
+    'Reduce_test': {'itr': 100000000, 'ranks': 4},
+    'Scan_test': {'itr': 100000000, 'ranks': 4},
+    'Scatter_test': {'itr': 100000000, 'ranks': 3},
+    'Scatter_test': {'itr': 100000000, 'ranks': 4},
+    'Sendrecv_test': {'itr': 100000000, 'ranks': 3},
+    'Testany_test': {'itr': 10, 'ranks': 3},
+    'Type_commit_contiguous': {'itr': 10, 'ranks': 4},
+    'Type_hvector_test': {'itr': 100000000, 'ranks': 2},
+    'Type_vector_test': {'itr': 100000000, 'ranks': 2},
+    'Waitall_test': {'itr': 10, 'ranks': 4},
+    'Waitany_test': {'itr': 10, 'ranks': 4},
+    'file_test': {'itr': 100000000, 'ranks': 2},
+    'keyval_test': {'itr': 12, 'ranks': 4},
+    'large_async_p2p': {'itr': 12, 'ranks': 2},
+    'send_recv_loop': {'itr': 12, 'ranks': 3},
+    'sendrecv_replace_test': {'itr': 100000000, 'ranks': 3},
+    'two-phase-commit-1': {'itr': 60, 'ranks': 4},
+    'two-phase-commit-2': {'itr': 60, 'ranks': 4}
+}
+
+# Custom argparser to eprint error message
+class CustomParser(argparse.ArgumentParser):
+    def error(self, message):
+        sys.stderr.write('error: %s\n' % message)
+        self.print_help()
+        os._exit(2)
+'''
+Example usage:
+
+python3 run_all.py -r -m ~/mana
+
+'''
+def main():
+    parser = CustomParser(description='Run all MANA Tests')
+    parser.add_argument('-m','--mana_root', metavar='M', help='Absolute \
+                         path to mana folder', default='', required=False)
+    parser.add_argument('-r','--mpirun', help='Use mpirun instead \
+                         of srun', action="store_true")
+
+    args = parser.parse_args()
+    mana = args.mana_root
+    if mana == '':
+        mana = os.path.abspath(__file__).replace('/mpi-proxy-split/'
+                                                 'test/run_all.py', '')
+    mpirun = ''
+    if args.mpirun:
+        mpirun = '-r'
+    num_tests = 0
+    num_passed = 0
+    failed = []
+
+    for test in tests:
+        print(f'Running test {test}')
+        i = tests[test]['itr']
+        n = tests[test]['ranks']
+        if 'args' in tests[test]:
+            a = tests[test]['args']
+            test_child=subprocess.run(['python3', f'{mana}/mpi-proxy-split'
+                                      '/test/run_mana_mpi_regression_test.py',
+                                      f'{mpirun}', '-m', f'{mana}', '-t', '120',
+                                      '-i', f'{i}', '-n', f'{n}', f'-a={a}',
+                                      f'{test}'],
+                                      stdout=subprocess.DEVNULL,
+                                      stderr=subprocess.DEVNULL)
+        else:
+            test_child=subprocess.run(['python3', f'{mana}/mpi-proxy-split'
```

Comment:
The newer changes broke the working version. This is related to the position of the test case.